packages/server/src/api/controllers/datasource.js

```diff
@@ -50,9 +50,21 @@ exports.fetch = async function (ctx) {
 exports.buildSchemaFromDb = async function (ctx) {
   const db = getAppDB()
   const datasource = await db.get(ctx.params.datasourceId)
+  const tablesFilter = ctx.request.body.tablesFilter
 
-  const { tables, error } = await buildSchemaHelper(datasource)
-  datasource.entities = tables
+  let { tables, error } = await buildSchemaHelper(datasource)
+  if (tablesFilter) {
+    if (!datasource.entities) {
+      datasource.entities = {}
+    }
+    for (let key in tables) {
+      if (tablesFilter.includes(key)) {
```

Comment:
this should probably be case insensitive - usually users don't expect to need to define a table name exactly (as some SQL databases treat identifiers as case insensitive).

Reply:
The only risk is you could technically pull down two tables when you only want one, e.g. "appointments":

![Screenshot 2022-09-13 at 10 37 34](https://user-images.githubusercontent.com/101575380/189867528-ed4a23ba-1359-412b-be57-c6b8f93491b4.png)

With that said maybe it's very unlikely users would have two tables with the same name except the casing.